disnake/ui/action_row.py

```diff
@@ -0,0 +1,236 @@
+        The items of this action row.
+    """
+
+    def __init__(self, *items: Item):
+        self._total_width = 0
+        components = []
+        # Validate the components
+        for item in items:
+            if not isinstance(item, Item):
+                raise ValueError("ActionRow must contain only Item instances")
+
+            self._total_width += item.width
+
+            if self._total_width > 5:
+                raise ValueError(
+                    "Too many items in 1 row. There should be not more than 5 buttons or 1 menu."
+                )
+
+            components.append(item._underlying)  # type: ignore
+
+        self._underlying = ActionRowComponent._raw_construct(
+            type=ComponentType.action_row,
+            children=components,
+        )
+
+    @property
+    def children(self) -> List[Component]:
+        """List[:class:`Component`]: The components of this row."""
+        return self._underlying.children
+
+    @children.setter
+    def children(self, value: List[Component]):
+        self._underlying.children = value
+
+    @property
+    def type(self) -> ComponentType:
+        return self._underlying.type
+
+    def append_item(self, item: Item) -> None:
+        """Appends an item to the action row.
+
+        Parameters
+        -----------
+        item: :class:`disnake.ui.Item`
+            The item to append to the action row.
+
+        Raises
+        -------
+        ValueError
+            The width of the action row exceeds 5.
+        """
+        if self._total_width + item.width > 5:
+            raise ValueError("Too many items in this row, can not appen a new one.")
+
+        self._total_width += item.width
+        self._underlying.children.append(item._underlying)  # type: ignore
+
+    def add_button(
+        self,
+        *,
+        style: ButtonStyle = ButtonStyle.secondary,
+        label: Optional[str] = None,
+        disabled: bool = False,
+        custom_id: Optional[str] = None,
+        url: Optional[str] = None,
+        emoji: Optional[Union[str, Emoji, PartialEmoji]] = None,
+    ):
+        """Adds a button to the action row.
+
+        To append a pre-existing :class:`disnake.ui.Button` use the
+        :meth:`append_item` method instead.
+
+        Parameters
+        -----------
+        style: :class:`disnake.ButtonStyle`
+            The style of the button.
+        custom_id: Optional[:class:`str`]
+            The ID of the button that gets received during an interaction.
+            If this button is for a URL, it does not have a custom ID.
+        url: Optional[:class:`str`]
+            The URL this button sends you to.
+        disabled: :class:`bool`
+            Whether the button is disabled or not.
+        label: Optional[:class:`str`]
+            The label of the button, if any.
+        emoji: Optional[Union[:class:`.PartialEmoji`, :class:`.Emoji`, :class:`str`]]
+            The emoji of the button, if available.
```

Comment:
I think "if any" and "if available" isn't necessary there, is it? (same question below in `add_select`)

Reply:
Perhaps